src/nupic/os/Directory.cpp

```diff
@@ -23,7 +23,9 @@
 /** @file 
 */
 
-#include <unistd.h>
+#if !defined(NTA_OS_WINDOWS)
+  #include <unistd.h>
```

Comment:
I like your suggestion of moving unistd.h include to same location as windows.h. It will make it more obvious what is going on.